src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPPollingSource.java

```diff
@@ -87,4 +122,159 @@ public void onTerminate() {
     endUsingFolder();
   }
 
+  @Override
+  protected Message[] getMessages(Folder openFolder) {
+    if (!getPredicateBuilder().isPresent() || !getPredicateBuilder().get().getEnableRemoteSearchFilter()) {
+      //Filters will be applied locally.
+      return super.getMessages(openFolder);
+    }
+
+    IMAPEmailPredicateBuilder matcher = getPredicateBuilder().orElseGet(() -> new IMAPEmailPredicateBuilder());
```

Comment:
I think that the searchTerm could be created once and reused instead of having to create it on each poll.